src/pybind/rados/rados.pyx

```diff
@@ -557,6 +557,11 @@ def decode_cstr(val, encoding="utf-8"):
     return val.decode(encoding)
 
 
+def flatten_dict(d):
```

Comment:
```
2021-05-07T04:44:28.499 INFO:tasks.workunit.client.0.smithi051.stderr:======================================================================
2021-05-07T04:44:28.500 INFO:tasks.workunit.client.0.smithi051.stderr:ERROR: test_rados.TestIoctx.test_service_daemon
2021-05-07T04:44:28.500 INFO:tasks.workunit.client.0.smithi051.stderr:----------------------------------------------------------------------
2021-05-07T04:44:28.501 INFO:tasks.workunit.client.0.smithi051.stderr:Traceback (most recent call last):
2021-05-07T04:44:28.501 INFO:tasks.workunit.client.0.smithi051.stderr:  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
2021-05-07T04:44:28.505 INFO:tasks.workunit.client.0.smithi051.stderr:    self.test(*self.arg)
2021-05-07T04:44:28.505 INFO:tasks.workunit.client.0.smithi051.stderr:  File "/home/ubuntu/cephtest/clone.client.0/src/test/pybind/test_rados.py", line 890, in test_service_daemon
2021-05-07T04:44:28.505 INFO:tasks.workunit.client.0.smithi051.stderr:    self.rados.service_daemon_register("laundry", name, metadata)
2021-05-07T04:44:28.506 INFO:tasks.workunit.client.0.smithi051.stderr:  File "rados.pyx", line 517, in rados.requires.wrapper.validate_func (/build/ceph-14.2.20-208-ge6c75acd/obj-x86_64-linux-gnu/src/pybind/rados/pyrex/rados.c:5046)
2021-05-07T04:44:28.506 INFO:tasks.workunit.client.0.smithi051.stderr:  File "rados.pyx", line 1545, in rados.Rados.service_daemon_register (/build/ceph-14.2.20-208-ge6c75acd/obj-x86_64-linux-gnu/src/pybind/rados/pyrex/rados.c:19326)
2021-05-07T04:44:28.506 INFO:tasks.workunit.client.0.smithi051.stderr:  File "rados.pyx", line 562, in rados.flatten_dict (/build/ceph-14.2.20-208-ge6c75acd/obj-x86_64-linux-gnu/src/pybind/rados/pyrex/rados.c:6223)
2021-05-07T04:44:28.507 INFO:tasks.workunit.client.0.smithi051.stderr:  File "rados.pyx", line 522, in rados.cstr (/build/ceph-14.2.20-208-ge6c75acd/obj-x86_64-linux-gnu/src/pybind/rados/pyrex/rados.c:5338)
2021-05-07T04:44:28.507 INFO:tasks.workunit.client.0.smithi051.stderr:TypeError: cstr() takes at least 2 positional arguments (1 given)
2021-05-07T04:44:28.507 INFO:tasks.workunit.client.0.smithi051.stderr:
2021-05-07T04:44:28.507 INFO:tasks.workunit.client.0.smithi051.stderr:----------------------------------------------------------------------
```
/ceph/teuthology-archive/yuriw-2021-05-06_15:23:19-rados-wip-yuri-testing-2021-05-05-1246-nautilus-distro-basic-smithi/6101414/teuthology.log

looks related, will verify with @tchaikov

Reply:
@ideepika can we verify if https://github.com/ceph/ceph/pull/41318 fixes the issue https://tracker.ceph.com/issues/50692?

Reply:
@neha-ojha sure, I see @yuriw included it in recent batch, will check